File: src/main/java/edu/marmara/model/Schedule.java
package edu.marmara.model;

import java.util.List;


public class Schedule {
    private List<Course> courses;
    private Boolean approved;
    private Boolean sendToReview;

    public Schedule() {
    }

    public Schedule(List<Course> courses, Boolean approved, Boolean sendToReview) {
        this.courses = courses;
        this.approved = approved;
        this.sendToReview = sendToReview;
    }

    @Override
    public String toString() {
        return "Schedule{" +
                "courses=" + courses +
                '}';
    }
    public void addCourse(Course course) {
        this.courses.add(course);
    }

    public void removeCourse(Course course) {this.courses.remove(course);}

    public Boolean getApproved() {
        return approved;
    }

    public void setApproved(Boolean approved) {
        this.approved = approved;
    }

    public Boolean getSendToReview() {
        return sendToReview;
    }

    public void setSendToReview(Boolean sendToReview) {
        this.sendToReview = sendToReview;
    }

    public List<Course> getCourses() {
        return courses;
    }

    public void setCourses(List<Course> courses) {
        this.courses = courses;
    }
}

File: src/python/src/main/marmara/model/Schedule.py
from typing import List

from src.main.marmara.model.Course import Course


class Schedule:
    def __init__(self, courses: List[Course], approved: bool, send_to_review: bool):
        self.courses = courses
        self.approved = approved
        self.send_to_review = send_to_review

    def __str__(self) -> str:
        return f"Schedule{{courses={self.courses}}}"

    def add_course(self, course: Course):
        self.courses.append(course)

    def remove_course(self, course: Course):
        self.courses.remove(course)

    @property
    def approved(self) -> bool:
        return self._approved

    @approved.setter
    def approved(self, value: bool):
        self._approved = value

    @property
    def send_to_review(self) -> bool:
        return self._send_to_review

    @send_to_review.setter
    def send_to_review(self, value: bool):
        self._send_to_review = value

    @property
    def courses(self) -> List[Course]:
        return self._courses

    @courses.setter
    def courses(self, value: List[Course]):
        self._courses = value
